### ISSUE TITLE
Why is context on taggable table?

### ISSUE THREAD

**Issue by NeoElit** [mbleigh/acts-as-taggable-on#700]
Why is context is entered each time in taggings table instead of putting it in the tags table?
Instead of repeating context on each tagging entry, would it be better if it is added and separate tags are created for different contexts?

**Comment by seuros** [comment]
because a tag can belong to many contexts.

**Comment by NeoElit** [comment]
yeah that right. But my query is, isn't it good to create tag table with context (different tag_id for unique [tag_name, tag_context]) instead of repeating string context in each tagging ? Normally tagging entries will be out numbered by tag entries.

**Comment by walidvb** [comment]
I agree with @NeoElit NeoElit, and have a usecase where i would rather have contexts in the `tags` table. I don't really see where i would want to share the same tag across contexts. 

I have a  `Book` and a `User` model. `Book`s are taggable_on `genres`, and `User`s on `fav_genres`.
By merging the two contexts, it is difficult to get only the genres of all `Book`s, or the count of all _history_  books(the count would also return the number of users who have _history_ as a `fav_genre`).

What is the usecase of the current implementation? If i have a user that has `skills = 'french'` **and** `languages='french'`, i would not want the tag `french` to have 2 counts, imo.

**Comment by digerata** [comment]
One of the things you can't do without context on the tags table is present a list of tags to users based on the context. Being able to do this is a huge usability win. E.g., "Here are tags you can use for hobbies and it won't show the tags used for movie genres."

(In fact, we maintain a fork of this library with the tags table having a context column.)

**Comment by razum2um** [comment]
@digerata please take a look on https://github.com/mbleigh/acts-as-taggable-on/pull/818 - will it solve issue for you?
